xtc/src/utils.rs

```diff
@@ -0,0 +1,21 @@
+use ic_kit::candid::Nat;
+use std::convert::TryFrom;
+
+pub fn convert_nat_to_u64(num: Nat) -> Result<u64, String> {
+    let u64_digits = num.0.to_u64_digits();
+
+    match u64_digits.len() {
+        0 => Ok(0),
+        1 => Ok(u64_digits[0]),
+        _ => Err("Nat -> Nat64 conversion failed".to_string()),
+    }
+}
+
+pub fn convert_nat_to_u32(num: Nat) -> Result<u32, String> {
```

Comment:
this could probably be a generic function working for uX, but we would still need to abstract away calls like: to_u64_digits(), to_u32_digits() to be able to write that generic function